fabtests/common/hmem_ze.c

```diff
@@ -274,9 +274,12 @@ int ft_ze_init(void)
 
 	count = 0;
 	ze_ret = (*libze_ops.zeDeviceGet)(driver, &count, NULL);
-	if (ze_ret || count > ZE_MAX_DEVICES)
+	if (ze_ret)
 		goto err;;
 
+	if (count > ZE_MAX_DEVICES)
```

Comment:
If you're going to allow it to continue and ignore the other devices, then we should probably add some checks in the copy function which uses the device number in case they use something out of bounds. We can still copy using a different device engine, just might not be optimized.
The shareable handle code might need some adjustment as well since we won't have the device fd for the memory. Not sure what to do about that... might just have to return an error and let shm fall back to bouncing through the host.

Reply:
Let's solve it all by removing the limit.

Reply:
A fast fix is fine, but I would continue to fail if count > max, because I'm not sure the other places in the code will handle things correctly.

Reply:
The statement about ze_hmem_init_fds() is not accurate, but it does need rework to account for the difference between number of all physical devices and the number of exposed devices. In addition, sub devices may have their own uuid.